src/eos-autoupdater.c

```diff
@@ -105,6 +106,86 @@ get_stamp_dir (void)
                         UPDATE_STAMP_DIR);
 }
 
+static void
+log_with_msgid (const gchar *msgid,
+                GLogLevelFlags log_level,
+                const gchar *numeric_log_level,
+                const gchar *format,
+                va_list args)
+{
+  g_autofree gchar *message = NULL;
+  /* Apparently the version of GCC in Endless ignores the
+   * G_GNUC_PRINTF annotation that has a zero as the second parameter,
+   * so it suggests to use this attribute. */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=format"
+  gint message_length = g_vasprintf (&message, format, args);
+#pragma GCC diagnostic pop
+  const GLogField fields[] = {
+    { "MESSAGE", message, message_length },
+    { "MESSAGE_ID", msgid, EOS_UPDATER_MSGID_LENGTH },
+    { "PRIORITY", numeric_log_level, 1 },
+    /* strlen(G_LOG_DOMAIN) should be folded into a constant at
+     * compilation time */
+    { "GLIB_DOMAIN", G_LOG_DOMAIN, (G_LOG_DOMAIN != NULL) ? strlen(G_LOG_DOMAIN) : 0 },
```

Comment:
I was rather wondering if I should rather drop all this dance around `G_LOG_DOMAIN` being NULL, since we set it during compilation, so the field could be always included. But for now I'll make it simpler like you suggested.